snapcraft/internal/meta/_desktop.py

```diff
@@ -35,8 +35,8 @@ def __init__(
         self._prime_dir = prime_dir
         self._path = os.path.join(prime_dir, filename)
         if not os.path.exists(self._path):
-            raise errors.InvalidDesktopFileError(
-                filename, 'does not exist (defined in the app {!r})'.format(
+            raise errors.MissingDesktopFileError(
+                filename, 'does not exist (defined in the app {!r}) \n Check if desktop file matches app name or Create a new path '.format(
```

Comment:
Passing a message in here is no longer necessary. Just pass the filename and the app name, and do the rest in the exception itself.